### ISSUE TITLE
Challenge 1

### ISSUE THREAD

**Issue by vashuag** [scaleracademy/scaler-september-open-source-challenge#425]
name: Vashu Agarwal
github_user_name: vashuag

**Comment by vashuag** [comment]
Screenshot showing the list of all the current branches inside of your forked clone repository

<img width="583" alt="Screenshot 2022-09-03 at 10 32 15 PM" src="https://user-images.githubusercontent.com/83650895/188281101-d33f22ee-37b5-4b25-905e-1edf6a6ceb89.png">

**Comment by vashuag** [comment]
<img width="1048" alt="Screenshot 2022-09-04 at 10 05 09 PM" src="https://user-images.githubusercontent.com/83650895/188324310-30327127-9a26-479f-b52f-816aaa424c3c.png">
<img width="1048" alt="Screenshot 2022-09-04 at 10 12 39 PM" src="https://user-images.githubusercontent.com/83650895/188324323-df1b9b37-19ca-4a43-8f19-4ee851d0e2d6.png">

**Comment by vashuag** [comment]
<img width="965" alt="Screenshot 2022-09-10 at 12 38 59 AM" src="https://user-images.githubusercontent.com/83650895/189426453-6571f090-0eab-4639-a6d1-9f59f54b3aed.png">

<img width="1331" alt="Screenshot 2022-09-10 at 12 40 59 AM" src="https://user-images.githubusercontent.com/83650895/189426337-8e860552-ac6e-4807-8ff8-b84e389a358f.png">
e.png">

**Comment by vashuag** [comment]
Challenge 11 completed

<img width="716" alt="Screenshot 2022-09-11 at 7 56 06 PM" src="https://user-images.githubusercontent.com/83650895/189533291-65dfdfe8-d838-40e8-bf3a-64f53b6d790d.png">

**Comment by vashuag** [comment]
<img width="930" alt="Screenshot 2022-09-13 at 11 22 54 PM" src="https://user-images.githubusercontent.com/83650895/189975252-35b0a6cf-018d-40fa-8861-4ba2411c8ba0.png">
Challenge 13 completed

**Comment by DaltonDeven** [comment]
Thanks for putting in the effort we appreciate it very much, you can continue to take part in the challenge , but wont be eligible for the rewards because challenge 9,10 is not done correctly and there are lot of commits in PR

**Comment by vashuag** [comment]
> Thanks for putting in the effort we appreciate it very much, you can continue to take part in the challenge , but wont be eligible for the rewards because challenge 9,10 is not done correctly and there are lot of commits in PR

Thank you, for letting me know my status.
I have read on your terms, that if I complete the challenge for complete 30 days then  irrespective of the issues , i will be eligible for rewards.

**Comment by vashuag** [comment]
<img width="809" alt="Screenshot 2022-09-14 at 11 48 39 PM" src="https://user-images.githubusercontent.com/83650895/190233760-ed4584ff-40d8-463d-b57f-cc869f563977.png">


<img width="809" alt="Screenshot 2022-09-14 at 11 49 00 PM" src="https://user-images.githubusercontent.com/83650895/190233772-1cd30720-7c57-43bb-909b-7bf4cdcbc364.png">


<img width="830" alt="Screenshot 2022-09-14 at 11 53 53 PM" src="https://user-images.githubusercontent.com/83650895/190233778-45e70541-0b91-422d-b264-c8eeb747e036.png">
Completed challenge 14 with merged and resolved conflicts

**Comment by vashuag** [comment]
Thank you, for letting me know my status.I have read on your terms, that if
I complete the challenge for complete 30 days then irrespective of the
issues , i will be eligible for rewards. Please let me know will i get any
rewards later on . Although irrespective of that these 30 days are going to
very knowledgable , thank you for opportunity.

Vashu Agarwal
github username :vashuag

**Comment by vashuag** [comment]
<img width="1040" alt="Screenshot 2022-09-17 at 4 28 31 AM" src="https://user-images.githubusercontent.com/83650895/190828347-daf08832-bc60-4cbe-a8b8-5efdb17209d6.png">
Resolved conflict, Challenge 16 completed

**Comment by vashuag** [comment]
<img width="1040" alt="Screenshot 2022-09-17 at 4 28 31 AM" src="https://user-images.githubusercontent.com/83650895/190871490-40715047-f099-4e7e-ac2d-b5cf9860822d.png">
<img width="960" alt="Screenshot 2022-09-17 at 11 26 13 PM" src="https://user-images.githubusercontent.com/83650895/190871497-6374785e-bde0-4845-ac68-dc94e6d0865c.png">
<img width="960" alt="Screenshot 2022-09-17 at 11 26 49 PM" src="https://user-images.githubusercontent.com/83650895/190871501-08b882a6-bec6-464c-a613-76215872a3c9.png">
<img width="960" alt="Screenshot 2022-09-17 at 11 36 05 PM" src="https://user-images.githubusercontent.com/83650895/190871502-e6effe52-867e-4558-a28b-157015ee5a2b.png">
<img width="1416" alt="Screenshot 2022-09-17 at 11 39 30 PM" src="https://user-images.githubusercontent.com/83650895/190871504-3253d6a1-91dd-4c7c-940c-bd7214b47535.png">
<img width="1416" alt="Screenshot 2022-09-17 at 11 57 08 PM" src="https://user-images.githubusercontent.com/83650895/190871505-8a0fcd33-ba2f-4566-a2bd-e5047c57c098.png">
using git hub desktop and performing commit , merge ,push , pull

**Comment by vashuag** [comment]
Retrying challenge 9,10 as given in feedback
<img width="1017" alt="Screenshot 2022-09-18 at 11 13 16 PM" src="https://user-images.githubusercontent.com/83650895/190922035-a2fc309d-6eda-4fa1-9766-ce07e5862315.png">
<img width="642" alt="Screenshot 2022-09-18 at 11 14 02 PM" src="https://user-images.githubusercontent.com/83650895/190922037-485c5330-3330-4935-aae9-7e368f51fb21.png">

**Comment by vashuag** [comment]
<img width="1426" alt="Screenshot 2022-09-20 at 6 40 28 PM" src="https://user-images.githubusercontent.com/83650895/191266705-a603ac1c-f1e0-421a-8561-151a9bb55acf.png">
challenge 20 using project board on github

**Comment by vashuag** [comment]
<img width="872" alt="Screenshot 2022-09-21 at 7 20 49 PM" src="https://user-images.githubusercontent.com/83650895/191522559-3a36a325-de8d-435a-81d0-46a929ff7f63.png">
<img width="872" alt="Screenshot 2022-09-21 at 7 21 26 PM" src="https://user-images.githubusercontent.com/83650895/191522592-55dcdc4f-0969-48ed-857b-1cbbd7f9d662.png">
challenge 21 making git alias commands and using it

**Comment by vashuag** [comment]
<img width="801" alt="Screenshot 2022-09-22 at 11 01 36 AM" src="https://user-images.githubusercontent.com/83650895/191666732-9819b316-845f-4ab3-9b6e-660145de9da8.png">
<img width="1255" alt="Screenshot 2022-09-22 at 11 01 52 AM" src="https://user-images.githubusercontent.com/83650895/191666743-7154d077-aa8b-4551-873a-51e8021c1f2e.png">
Redone challenge 20

**Comment by kuraakhilesh8230** [comment]
Congrats you have successfully completed milestone 2 and you are good to move onto next milestone
All the best✅️

**Comment by vashuag** [comment]
Challenge 23


>>Things to be done for effective Open Source Contribution:

Follow Standard Git Branching Naming Conventions , such as;
start branch name with Group word •	bug-logo-alignment-issue 
Using unique id : wip-8712-add-testing-module  which makes possible to track the progress from external system
Using Hyphen ,Slash or Underscore as Separators : feature_upgrade_jquery_version_login_module
Using Author Name : <author>_<branch-type>_<branch-name>
Things to avoid while naming branches:
Using Numbers only ,using all naming conventions simultaneously
Making long descriptive names long-lived branches

>>For Making a better Pull Request

First and foremost , using above things for giving feature  branch a clear name.
Giving all PRs and commits a active and descriptive title  and use present imperative tense
Pls should be with meaningful description with 
Why the code necessary? How you are going to solve ? and What functionality added  or changed?
Whenever possible showing functionality visually  giving reviewer the  gist of work beforehand
The most important thing is to review our own PR before assigning to others with running tests, linting and to caught up with latest changes

>>Working with issue
Make sure to read labels properly some of the mean  the following  ;
help wanted: Open to participation from the community but not necessarily beginner-friendly
good first issue: Open to participation from the community and friendly towards new contributors
staff only: Requires infrastructure access or institutional knowledge

>>issues not ready to work
status: blocked: Blocked by other work that needs to be done first
status: ticket work required: Needs additional work before it is ready to be taken up
status: awaiting triage: Has not been triaged by a maintainer

If there is no label , add comment asking the maintainers to triage the issue .

Before actually starting work with any issue , check for comments /labels to see whether someone else has indicated that they are working on it

>>The Overall Contribution Process
Comment on issue on which we are working
write code and submit pull request
wait for code review and address issue raised as soon as we can

>>Proposing a new issue
1.	Create a new GitHub issue associated with the relevant repository and propose our change there .
2.	The issue will automatically have the 🚦 status: awaiting triage label applied. Wait for a project maintainer to evaluate your issue and decide whether it's something that we will accept a pull request for.
3.	Once the project maintainer has approved the issue and removed the 🚦 status: awaiting triage label, we may start work on code as described in the "Contribution process" section above.

**Comment by vashuag** [comment]
Challenge 24
Still looking for beginner friendly issue which I can feel confident to make changes in.Scrolled through various repos ,issues from the link provided in challenge https://goodfirstissue.dev/language/

**Comment by vashuag** [comment]
Challenge 25
Have looked for some beginner freindly issues on different repos and have commented on few to work on . waiting for responses and looking for registering on Hacktoberfest

**Comment by vashuag** [comment]
Registered for Hacktober fest and started contributing to open source in beginner friendly projects

**Comment by vashuag** [comment]
Challenge 26
Registered for Hacktober fest and started contributing to open source in beginner friendly projects

**Comment by vashuag** [comment]
Challenge 27
Some most important tips and rules are as follows:

1. The PRs and MRs should be in bound dates from 1 oct to 31 oct.
2. Anyone should not spam in any repos
3. Contributions with metioned /tagged "Hacktober fest " will be counted only.
4. Certain links for beginner friendly issues are given :https://github.com/firstcontributions/first-contributions
5. Follow Standard Git Branching Naming Conventions , such as;
start branch name with Group word • bug-logo-alignment-issue
Using unique id : wip-8712-add-testing-module which makes possible to track the progress from external system
Using Hyphen ,Slash or Underscore as Separators : feature_upgrade_jquery_version_login_module
Using Author Name : 
Things to avoid while naming branches:
Using Numbers only ,using all naming conventions simultaneously
Making long descriptive names long-lived branches

6.For Making a better Pull Request
First and foremost , using above things for giving feature branch a clear name.
Giving all PRs and commits a active and descriptive title and use present imperative tense
Pls should be with meaningful description with
Why the code necessary? How you are going to solve ? and What functionality added or changed?
Whenever possible showing functionality visually giving reviewer the gist of work beforehand
The most important thing is to review our own PR before assigning to others with running tests, linting and to caught up with latest changes

**Comment by vashuag** [comment]
Challenge 28

Tried a simple contribution of adding a card with full stack resources and merge successfully and under review .
link to pr:  https://github.com/Syknapse/Contribute-To-This-Project/pull/1702
<img width="1318" alt="Screenshot 2022-09-29 at 1 00 35 PM" src="https://user-images.githubusercontent.com/83650895/192968218-11360c30-21a1-4727-af09-4f032de54ed7.png">

**Comment by vashuag** [comment]
Challenge 29
<img width="1403" alt="Screenshot 2022-09-30 at 2 03 52 PM" src="https://user-images.githubusercontent.com/83650895/193228520-a952e99f-37d4-4fff-b7e8-5172ea283956.png">
<img width="1403" alt="Screenshot 2022-09-30 at 2 04 03 PM" src="https://user-images.githubusercontent.com/83650895/193228539-b9fe07eb-e54d-497e-be58-0f774e5c3875.png">

Started a new discussion with idea of Ai Drones and upvoted some of the ideas I liked

**Comment by vashuag** [comment]
Challenge 30 
Last Day of this Scaler Open Source Challenge , I literally learned and enjoyed a lot , in this event .Thank you Scalar for providing such an amazing opportunity just before the Hacktoberfest, which has given me confidence to contribute to open source on my own .Thank you @kuraakhilesh8230 @DaltonDeven @shivaylamba for checking and verifying my milestones , and giving me feedback for it . I will surely encourage others to contribute to open source and make world the better place ,and this community to grow

**Comment by Mrjoy832** [comment]
## Milestone-4 done ✅
